src/brod_client.erl

```diff
@@ -206,14 +248,18 @@ handle_info({'EXIT', Pid, Reason},
                         [ClientId, Host, Port, Reason]),
   NewSock = start_metadata_socket(ClientId, Endpoints),
   {noreply, State#state{meta_sock = NewSock}};
-handle_info({'EXIT', Pid, Reason}, State) ->
+handle_info({'EXIT', Pid, Reason}, #state{meta_sock = Pid} = State) ->
```

Comment:
meta_sock is a #sock{} record.
and, it's handled in handle_info clause line 240